modules/game.feature.menu.menu-model.d.ts

```diff
@@ -167,30 +167,109 @@ declare global {
       hotkeysCallbacks: sc.MenuModel.HotkeyCallback;
       currentBackCallback: sc.MenuModel.BackCallback;
       buttonInteract: ig.ButtonInteractEntry;
-      shopID: string | null | undefined;
+      shopID: Optional<string>;
       shopPage: number;
       shopCart: sc.MenuModel.ShopCartEntry[];
       shopSellMode: boolean;
       statusElement: sc.ELEMENT;
       statusDiff: boolean;
+      previousMenu: sc.MENU_SUBMENU;
```

Comment:
Additionally, `backCallbackStack` should be under `buttonInteract` but I could not add it to the suggestion because the hunk is outside this diff
```suggestion
      previousMenu: sc.MENU_SUBMENU;
      buttonInteract: ig.ButtonInteractEntry;
      hotkeysCallbacks: sc.MenuModel.HotkeyCallback;
      currentBackCallback: sc.MenuModel.BackCallback;
      shopID: Optional<string>;
      shopPage: number;
      shopCart: sc.MenuModel.ShopCartEntry[];
      shopSellMode: boolean;
      statusElement: sc.ELEMENT;
      statusDiff: boolean;
```